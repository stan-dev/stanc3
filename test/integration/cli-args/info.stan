#include <included.stan>
#include <recursive.stan>

functions {
  real foo(real a) {
    return sin(a);
  }
  real goo_lpdf(real a) {
    return a;
  }
  real f_lpdf(real[] y_slice, int start, int end) {
    return normal_lpdf(y_slice| 0, 1);
  }
}

data {
  int a;
  real b;
  vector[2] c;
  row_vector[3] d;
  matrix[2,2] e;
  int f[5];
  real g[6];
  vector[1] h[7];
  matrix[2,2] i[2];
  int j[3,1,3];
}

transformed data {
  int k = a + 1;
}

parameters {
  simplex[10] l;
  unit_vector[11] m;
  ordered[12] n;
  positive_ordered[13] o;
  cov_matrix[14] p;
  corr_matrix[15] q;
  cholesky_factor_cov[16] r;
  cholesky_factor_corr[17] s;
  real y;
}

transformed parameters {
  matrix[14, 14] t = p;
  {
    real ignored_tp = log(r[0][0]);
  }
}

model {
    real ignored_model = square(foo(square(r[0][0])));
    l ~ dirichlet(c);
    target += normal_lcdf(y| 0, 1);
    target += normal_lccdf(y| 0, 1);
    target += std_normal_lupdf(y);
    target += reduce_sum(f_lpdf, g, 1);
    y ~ goo();
}

generated quantities {
  real u = l[0];
  {
    real ignored_gq = l[1];
  }
}
