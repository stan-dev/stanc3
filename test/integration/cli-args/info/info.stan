#include <included.stan>
#include <recursive.stan>

functions {
  real foo(real a) {
    return sin(a);
  }
  real goo_lpdf(real a) {
    return a;
  }
  real f_lpdf(array[] real y_slice, int start, int end) {
    return normal_lpdf(y_slice| 0, 1);
  }
}

data {
  int a;
  real b;
  vector[2] c;
  row_vector[3] d;
  matrix[2,2] e;
  array[5] int f;
  array[6] real g;
  array[7] vector[1] h;
  array[2] matrix[2,2] i;
  array[3,1,3] int j;
}

transformed data {
  int k = a + 1;
}

parameters {
  simplex[10] l;
  unit_vector[11] m;
  ordered[12] n;
  positive_ordered[13] o;
  cov_matrix[14] p;
  corr_matrix[15] q;
  cholesky_factor_cov[16] r;
  cholesky_factor_corr[17] s;
  real y;
}

transformed parameters {
  matrix[14, 14] t = p;
  {
    real ignored_tp = log(r[0][0]);
  }
}

model {
    real ignored_model = square(foo(square(r[0][0])));
    l ~ dirichlet(c);
    target += normal_lcdf(y| 0, 1);
    target += normal_lccdf(y| 0, 1);
    target += std_normal_lupdf(y);
    target += reduce_sum(f_lpdf, g, 1);
    y ~ goo();
}

generated quantities {
  real u = l[0];
  {
    real ignored_gq = l[1];
  }
}
