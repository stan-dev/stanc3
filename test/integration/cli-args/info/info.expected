  $ ../../../../../install/default/bin/stanc --include-paths=.,includes --info  info.stan
{
  "inputs": {
    "a": { "type": "int", "dimensions": 0 },
    "b": { "type": "real", "dimensions": 0 },
    "c": { "type": "real", "dimensions": 1 },
    "d1": { "type": "real", "dimensions": 1 },
    "d2": { "type": "real", "dimensions": 1 },
    "e": { "type": "real", "dimensions": 2 },
    "f": { "type": "int", "dimensions": 1 },
    "g": { "type": "real", "dimensions": 1 },
    "h": { "type": "real", "dimensions": 2 },
    "i": { "type": "real", "dimensions": 3 },
    "j": { "type": "int", "dimensions": 3 },
    "cplx": { "type": "complex", "dimensions": 1 },
    "cplx_vec": { "type": "complex", "dimensions": 1 },
    "cplx_row": { "type": "complex", "dimensions": 1 },
    "cplx_mat": { "type": "complex", "dimensions": 2 },
    "tuples": {
      "type": [
        { "type": "int", "dimensions": 0 },
        { "type": "real", "dimensions": 0 },
        {
          "type": [
            { "type": "real", "dimensions": 0 },
            { "type": "int", "dimensions": 1 }
          ],
          "dimensions": 0
        }
      ],
      "dimensions": 1
    }
  },
  "parameters": {
    "l": { "type": "real", "dimensions": 1 },
    "m": { "type": "real", "dimensions": 1 },
    "n": { "type": "real", "dimensions": 1 },
    "o": { "type": "real", "dimensions": 1 },
    "p": { "type": "real", "dimensions": 2 },
    "q": { "type": "real", "dimensions": 2 },
    "r": { "type": "real", "dimensions": 2 },
    "s": { "type": "real", "dimensions": 2 },
    "y": { "type": "real", "dimensions": 0 }
  },
  "transformed parameters": { "t": { "type": "real", "dimensions": 2 } },
  "generated quantities": { "u": { "type": "real", "dimensions": 0 } },
  "functions": [ "log", "print", "reduce_sum", "reject", "sin", "square" ],
  "distributions": [
    "dirichlet_lupdf", "normal_lccdf", "normal_lcdf", "normal_lpdf",
    "std_normal_lupdf"
  ],
  "included_files": [
    "./included.stan", "includes/recursive.stan",
    "includes/another_include.stan"
  ]
}
